lua/entities/gmod_wire_expression2/core/e2lib.lua

```diff
@@ -296,6 +296,19 @@ end
 
 -- ----------------------------- compiler stuf ---------------------------------
 
+-- A function suitable for use as xpcall's error handler. If the error is
+-- generated by Compiler:Error, Parser:Error, etc., then the string will be a
+-- usable error message. If not, then it's an error not caused by an error in
+-- user code, and so we dump a stack trace to the console to help debug it.
+function E2Lib.errorHandler(message)
+	if string.match(message, " at line ") then return message end
```

Comment:
http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/Throwable.java#Throwable.printStackTrace%28java.io.PrintStream%29
here's the code for that